### PR TITLE
add assertions on whether elementwise_div divison is zero

### PR DESCRIPTION
**Pull request by danleifeng**
throw exception for `int32,int64` type
### cpu:
```
PaddleCheckError: InvalidArgumentError: Integer division by zero encountered in divide.Please check.
```
### gpu:
throw expection in cuda kernel. 
```
Exception: ./elementwise/elementwise_op_function.cu.h:60 Assertion `b != 0` failed. InvalidArgumentError: Integer division by zero encountered in divide.Please check.
----------------------
Error Message Summary:
----------------------
PaddleCheckError: cudaMemcpy failed in paddle::platform::GpuMemcpySync (0x7fce60000240 -> 0x7fce2704a080, length: 32) error code : 4, Please see detail in https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html#group__CUDART__TYPES_1g3f51e3575c2178246db0a94a430e0038: unspecified launch failure at 
[./Paddle/paddle/fluid/platform/gpu_info.cc:261]
  [operator < fetch > error]
```

### CODE REVIEW

File: paddle/fluid/operators/elementwise/elementwise_div_op.cc [hunk 1]
```diff
@@ -32,6 +33,20 @@ struct SameDimsElemwiseDiv<
...
+    platform::CPUDeviceContext, T,
+    typename std::enable_if<std::is_integral<T>::value>::type> {
+  void operator()(const framework::ExecutionContext &ctx,
+                  const framework::Tensor *x, const framework::Tensor *y,
+                  framework::Tensor *z) {
+    default_elementwise_div<platform::CPUDeviceContext, T>(ctx, x, y, z);
+  }
+};
+
+/*
```
Comment (gongweibao) [inline paddle/fluid/operators/elementwise/elementwise_div_op.cc]: cleanup this!

File: paddle/fluid/operators/elementwise/elementwise_op_function.cu.h [hunk 2]
```diff
@@ -46,8 +52,18 @@ DEFINE_SIMPLE_BINARY_FUNCTOR(Mul, *)
...
+                  typename std::enable_if<std::is_integral<T>::value>::type> {
+  inline HOSTDEVICE T operator()(const T& a, const T& b) const {
+    PADDLE_ENFORCE(b != 0, DIV_ERROR_INFO);
+    return a / b;
+  }
+};
+
 #define DEFINE_SIMPLE_CUDA_BINARY_FUNCTOR(Func, expr)                         \
-  template <typename T>                                                       \
+  template <typename T, class Enable = void>                                  \
```
Comment (gongweibao) [inline paddle/fluid/operators/elementwise/elementwise_op_function.cu.h:66]: Where `enable` is used?

**Review by gongweibao — CHANGES_REQUESTED** [review]

File: paddle/fluid/operators/elementwise/elementwise_div_op.cc (hunk 1 above)
Comment (danleifeng) [reply paddle/fluid/operators/elementwise/elementwise_div_op.cc]: done.

File: paddle/fluid/operators/elementwise/elementwise_op_function.cu.h (hunk 2 above)
Comment (danleifeng) [reply paddle/fluid/operators/elementwise/elementwise_op_function.cu.h:66]: Enable is used for template specialization.  you can read line 86.

**Review by gongweibao — DISMISSED** [review]
LGTM

**Review by luotao1 — APPROVED** [review]

**Pull request merged**